recirq/quantum_chess/quantum_board.py

```diff
@@ -133,6 +147,15 @@ def undo_last_move(self) -> bool:
                 return False
         return True
 
+    def record_time(self, action: str, t0: float, t1: Optional[float] = None) -> None:
+        """Writes time span from t0 to t1 (if specified, otherwise the current time is used)
+        into the debug log and timing stats.
+        """
+        if t1 is None:
+            t1 = time.perf_counter()
+        self.debug_log += (f"{action} takes {t1 - t0:0.4f} seconds.\n")
+        self.timing_stats.append(t1 - t0)
```

Comment:
Should we append a tuple of action/time instead?  (or alternatively, make timing_stats a dictionary of action to list of times?)

Reply:
Thanks Doug! Yes that makes much sense. timing_stats is changed to dictionary, and the test has changed accordingly.